File: algorithms/python/fibonacci-dp.py
import sys
import time

MOD = 1_000_000_007

def fib_dp(n):
    memo = [0] * (n + 1)
    if n >= 1: memo[1] = 1
    for i in range(2, n + 1):
        memo[i] = (memo[i-1] + memo[i-2]) % MOD
    return memo[n]

def main():
    n = int(sys.stdin.read().strip())

    start  = time.perf_counter()
    result = fib_dp(n)
    ms     = (time.perf_counter() - start) * 1000

    sys.stdout.write(f"{result}\n")
    sys.stderr.write(f"{ms:.3f}\n")

if __name__ == "__main__":
    main()

File: algorithms/lua/fibonacci-dp.lua
local MOD = 1000000007

local function fib_dp(n)
    local memo = {}
    memo[0] = 0
    if n >= 1 then memo[1] = 1 end
    for i = 2, n do
        memo[i] = (memo[i-1] + memo[i-2]) % MOD
    end
    return memo[n]
end

local function main()
    local n      = tonumber(io.read("*l"))
    local start  = os.clock()
    local result = fib_dp(n)
    local ms     = (os.clock() - start) * 1000

    print(result)
    io.stderr:write(string.format("%.3f\n", ms))
end

main()